File: CS/MainForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.Utils.Win;

namespace PopupContainerWithOKButton
{
    public partial class MainForm : Form
    {
        public MainForm()
        {
            InitializeComponent();
        }

        private void popupContainerEditOKButtonpopupContainerEditOKButtonForTest_Properties_QueryResultValue(object sender, DevExpress.XtraEditors.Controls.QueryResultValueEventArgs e)
        {
            e.Value = buttonEdit1.EditValue;
        }

        private void checkEditShowOk_Properties_EditValueChanged(object sender, EventArgs e)
        {
            popupContainerEditOKButtonForTest.Properties.ShowOkButton = (bool)checkEditShowOk.EditValue;
            Control currentPopupWindow = (popupContainerEditOKButtonForTest as IPopupControl).PopupWindow;
            if (currentPopupWindow != null) currentPopupWindow.Dispose();
        }

        private void MainForm_Load(object sender, EventArgs e)
        {
            checkEditShowOk.EditValue = true;
        }

    }
}

File: VB/MainForm.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.Utils.Win

Namespace PopupContainerWithOKButton

    Public Partial Class MainForm
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub popupContainerEditOKButtonpopupContainerEditOKButtonForTest_Properties_QueryResultValue(ByVal sender As Object, ByVal e As DevExpress.XtraEditors.Controls.QueryResultValueEventArgs)
            e.Value = buttonEdit1.EditValue
        End Sub

        Private Sub checkEditShowOk_Properties_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs)
            popupContainerEditOKButtonForTest.Properties.ShowOkButton = CBool(checkEditShowOk.EditValue)
            Dim currentPopupWindow As Control = TryCast(popupContainerEditOKButtonForTest, IPopupControl).PopupWindow
            If currentPopupWindow IsNot Nothing Then currentPopupWindow.Dispose()
        End Sub

        Private Sub MainForm_Load(ByVal sender As Object, ByVal e As EventArgs)
            checkEditShowOk.EditValue = True
        End Sub
    End Class
End Namespace
